fuse/opcode.go

```diff
@@ -99,6 +99,9 @@ func doInit(server *Server, req *request) {
 		CongestionThreshold: uint16(server.opts.MaxBackground * 3 / 4),
 		MaxBackground:       uint16(server.opts.MaxBackground),
 	}
+	if server.opts.MaxReadAhead != 0 {
+		out.MaxReadAhead = uint32(server.opts.MaxReadAhead)
```

Comment:
Yes you are right (duh!) - go code should read

```go
if server.opts.MaxReadAhead != 0 &&  uint32(server.opts.MaxReadAhead) < out.MaxReadAhead {
		out.MaxReadAhead = uint32(server.opts.MaxReadAhead)
}
```

Doing this will make it impossible to set a max read ahead of larger than the default value that fuse provides (128k in my tests).

I had a look at the kernel code and it limits the max readahead to this initial value too.

So if you can't set the readahead to larger than the initial value then I can't see the point of this PR any more as I wanted to increase the max read ahead beyond the default value not decrease it!

Anyway I can make that fix and resubmit if you want.